common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -465,7 +470,11 @@ public static void copyMemory(long srcAddr, byte[] dst, int dstIndex, long lengt
      * consumer (one thread!).
      */
     public static <T> Queue<T> newMpscQueue() {
-        return new MpscLinkedQueue<T>();
+        if (hasUnsafe()) {
+            return JAVA_VERSION < 8 ? new MpscLinkedQueue7<T>() : new MpscLinkedQueue8<T>();
```

Comment:
@normanmaurer 

I always thought Netty 4 was for JDKD 6 because of home page comment:

**No additional dependencies, JDK 5 (Netty 3.x) or 6 (Netty 4.x) is enough**